packages/wonder-blocks-data/components/data.js

```diff
@@ -76,7 +76,7 @@ export default class Data<TOptions, TData> extends React.Component<
          * or we got new data/error.
          */
         if (!this._propsMatch(nextProps)) {
-            const newState = this._buildStateAndfulfillNeeds();
+            const newState = this._buildStateAndfulfillNeeds(nextProps);
```

Comment:
This was the issue. It was using the current props to determine state, instead of the props we were moving to. A silly silly mistake on my part.